packages/lib-sourcify/src/lib/CheckedContract.ts

```diff
@@ -756,38 +771,31 @@ function getDiffPositions(original: string, modified: string): number[] {
   return differences;
 }
 
-// Checks if a substring exists in the bytecode at a given position
+/**
+ *   Checks the raw bytecode indeed includes the auxdata diff at the given position
+ */
 function bytecodeIncludesAuxdataDiffAt(
   bytecode: string,
-  {
-    real,
-    diff,
-    offsetStart,
-    offsetEnd,
-  }: { real: string; diff: string; offsetStart: number; offsetEnd: number },
+  auxdataDiff: AuxdataDiff,
   position: number
 ): boolean {
-  const extracted = bytecode.substr(
-    position - offsetStart,
-    offsetStart + diff.length + offsetEnd
-  );
+  const { real, diffStart } = auxdataDiff;
+  // QUESTION: Can't we just use `real` instead of offSetEnd etc.?
+  const extracted = bytecode.slice(position - diffStart, real.length);
   return extracted === real;
 }
 
 function getAuxdatasDiff(originalAuxdatas: string[], editedAuxdatas: string[]) {
-  const auxdatasDiffs = [];
+  const auxdatasDiffs: AuxdataDiff[] = [];
   for (let i = 0; i < originalAuxdatas.length; i++) {
     const diffPositions = getDiffPositions(
       originalAuxdatas[i],
       editedAuxdatas[i]
     );
     auxdatasDiffs.push({
-      offsetStart: diffPositions[0],
-      offsetEnd:
-        originalAuxdatas[i].length -
-        diffPositions[diffPositions.length - 1] -
-        1,
       real: originalAuxdatas[i],
+      diffStart: diffPositions[0],
+      // QUESTION: Shouldn't this be editedAuxdatas[i]?
       diff: originalAuxdatas[i].substring(
```

Comment:
Shouldn't this be `editedAuxdatas[i]`

Reply:
When I later compare it here https://github.com/ethereum/sourcify/blob/a49b431c75da154a3eb2b40c86f92f9985384072/packages/lib-sourcify/src/lib/CheckedContract.ts#L818 I'm using the original bytecode as comparison so it's right the `originalAuxdatas[i]`